File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int[] hashSet = new int[26];
        for(int i=0; i<s.length(); i++){
            hashSet[s.charAt(i)-'a']++;
        }
        for(int i=0; i<s.length(); i++){
            if(hashSet[s.charAt(i) - 'a'] == 1)
            return i;
        }
        return -1;
    }
}

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int>hashSet(256, 0);
        for(int i=0; i<s.size(); i++){
            hashSet[s[i]]++;
        }
        for(int i=0; i<s.size(); i++){
            if(hashSet[s[i]] == 1)
            return i;
        }
        return -1;
    }
};